app/models/cherrypick_task.rb

```diff
@@ -11,236 +11,41 @@
-        add_control_request(batch, control_asset)
-      end
-      add_any_consecutive_control_requests(control_posns, batch, control_assets)
-    end
-  end
 
+  #
   # Returns a list with the destination positions for the control wells distributed by
   # using batch_id and num_plate as position generators.
-  def control_positions(batch_id, num_plate, total_wells, num_control_wells)
-    unique_number = batch_id
-
-    # Generation of the choice
-    positions = []
-    available_posns = available_control_positions(total_wells)
-
-    while positions.length < num_control_wells
-      current_size = available_posns.length
-      position = available_posns.slice!(unique_number % current_size)
-      position_for_plate = position % total_wells
-      positions.push(position_for_plate)
-      unique_number /= current_size
+  # @note wells_to_leave_free was originally hardcoded for 96 well plates at 24, in order to avoid
+  # control wells being missed in cDNA quant QC. This requirement was removed in
+  # https://github.com/sanger/sequencescape/issues/2967 however I've avoided stripping out the behaviour
+  # completely in case controls are used in other pipelines.
+  #
+  # @param batch_id [Integer] The id of the batch, used to generate a starting position
+  # @param num_plate [Integer] The plate number within the batch
+  # @param total_wells [Integer] The total number of wells on the plate
+  # @param num_control_wells [Integer] The number of control wells to lay out
+  # @param wells_to_leave_free [Integer] The number of wells to leave free at the front of the plate
+  #
+  # @return [Array<Integer>] The indexes of the control well positions
+  #
+  def control_positions(batch_id, num_plate, total_wells, num_control_wells, wells_to_leave_free: 0)
+    total_available_positions = total_wells - wells_to_leave_free
+    control_separation = total_available_positions / num_control_wells
+
+    raise StandardError, 'More controls than free wells' if num_control_wells > total_available_positions
+
+    Array.new(num_control_wells) do |control_well_index|
+      # The position of controls varies from batch to batch, and advances by one
+      # position for each plate in that batch. If we have multiple controls on a
+      # plate, we try to space them evenly. This establishes the relative
+      # positions for each control. We then take the modulo against the number
+      # of available wells to set the final position on the plate itself
+      position_index = batch_id + # We advance the control position as the batch_id increases
+                       num_plate - # Then once per plate in the batch
+                       (control_well_index * control_separation) # Finally we distribute the controls evenly
+      # Finally we layout the controls on the actual available positions on our plate
+      wells_to_leave_free + (position_index % total_available_positions)
     end
```

Comment:
Current method spreads the controls better across the plate, however it has the drawback that the results have less variability, for example:

With this version:
```
irb(main):023:0> CherrypickTask.new.control_positions(0,0,96,3, wells_to_leave_free: 0)
=> [0, 64, 32]
irb(main):024:0> CherrypickTask.new.control_positions(96,0,96,3, wells_to_leave_free: 0)
=> [0, 64, 32]
irb(main):026:0> CherrypickTask.new.control_positions(192,0,96,3, wells_to_leave_free: 0)
=> [0, 64, 32]
```

If using same conditions, all controls will be placed in exactly the same position after every 96 batches.  The "cycle" now is in the relation between batch_id and num_of_wells.
In previous version clash would only happened after a full cycle of all controls across the plate has happened.
For example, with a 96 well plate that we want to put 3 controls but leaving the first 3 columns empty:
-  If the plate needs to have 3 columns empty, that leaves: 96 wells - 24 (3 columns) = 72.
- Allocating 3 controls means 72, 71 and 70 free places when putting them in position; so the clash happens every ```72*71*70``` = 357840 batches

With previous version:
```
irb(main):014:0> CherrypickTask.new.control_positions(0,0,96,3)
=> [24, 25, 26]
irb(main):039:0> CherrypickTask.new.control_positions(357840,0,96,3)
=> [24, 25, 26]
irb(main):040:0> CherrypickTask.new.control_positions(715680,0,96,3)
=> [24, 25, 26]
```

But all values in between are completely different. Even with that, clashes could still happen with some combinations of num_plate, but this is the best we could give for an even distribution of controls without using randomness. 
I like a lot your idea of splitting the plate in regions to place the controls with separation between each other and I was thinking, could this concept be merged with the distribution of batches? Could we split the the plate in 3 regions (if 3 controls) and use the modulus to place each individual control in their region but keeping the long distribution in place?

Reply:
That would mean, of course, that we will not be able to keep the value of 357840 when splitting in regions. But should be something around:
```
72 places / 3 controls = 24 locations
24*24*24=13824 batches
```

Reply:
Using your idea of placing controls in regions I think I found a way of generating the controls in different regions while keeping some variability in between the positions; the sequence starts with:
```
     1 [24, 48, 72]
     2 [25, 49, 73]
     3 [26, 50, 74]
     4 [27, 51, 75]
     5 [28, 52, 76]
...
```
and only repeats again after 13825:

```
...
 13824 [47, 70, 93]
 13825 [24, 48, 72]
 13826 [25, 49, 73]
 13827 [26, 50, 74]
 13828 [27, 51, 75]
...
```
I'll add this in another pull request as I cannot access your branch.